### ISSUE TITLE
FFTW Plugin File Not Found

### ISSUE THREAD

**Issue by jboyd23** [ArjunaCluster/ArjunaUsers#132]
> If you notice an issue on arjuna, please first search the existing issues and ensure
> that it has not been reported. If you notice a similar example, please comment on that issue.
>
> Instructions:
> 1) Update the issue's title with something informative
> 2) Fill out the form with request information
> 3) Delete all instructions (Lines that start with `>`) and unused parts of the form
> 4) Submit the issue by clicking "Submit New Issue"
>
> Tips for getting help
> - Quote your shell/log files with backticks. It makes reading them **much** easier
> - Include **both** the output of a command **and** the command itself
> - A minimal working example is a huge plus, we do not debug user code

## Basic Info
Your Name:  Jarrett Boyd
Your Andrew ID: jboyd@andrew.cmu.edu

## Where it happened
Job Ids: 126752
Node(s) on which the problem occurred:

## What Happened
Expected Behavior: Simulations ran as normal
Observed Behavior: sbatch: error: spank: x11.so: Plugin file not found

## Log Files
Location of Log file Showing the Error: Simulation terminated before starting so no log file generatd
Location of Script showing [Minimum Working Example]:

> Please also attach any logs and the submission script to this issue.
> If you don't have a log/ bash script, delete the examples below

#!/bin/bash
#SBATCH -J a_10ns_1245 # Job name
#SBATCH -n 28 # Number of total cores
#SBATCH -N 1 # Number of nodes
#SBATCH -A barati
#SBATCH -p cpu
#SBATCH --mem-per-cpu=2000 # Memory pool for all cores in MB
#SBATCH -e /home/jboyd/Phos_DNA/error_lammps.err
#SBATCH -o /home/jboyd/Phos_DNA/out_lammps.out
#SBATCH --mail-type=END,FAIL,BEGIN # Type of email notification- BEGIN,END,FAIL,ALL
#SBATCH --mail-user=jboyd@andrew.cmu.edu # Email to which notifications will be sent
#SBATCH -t 7-0:00
echo "Job started on `hostname` at `date`"
module load cuda/10.2.89 
module load gcc/11.2.0
mpirun -np 28 /home/jboyd/lammps-29Sep2021/build/lmp < Phos_memb20A2_Complete.in

echo " "
echo "Job Ended at `date`"

[Minimum Working Example]: https://en.wikipedia.org/wiki/Minimal_working_example

## What I've Tried
  
# I thought that the modules may have updated to a newer version, so I reran the bash file with different combinations of module that were available using >> module avail. This still resulted in the same error being generated.

> Thank you for filling out the issue template. We will do our best to help you

**Comment by abillscmu** [comment]
Please attach an error message if there was one, along with logs and the location of an MWE.

**Comment by abillscmu** [comment]
Please upload the script and error.

**Comment by kianpu34593** [comment]
The error:
```sbatch: error: spank: x11.so: Plugin file not found```

Tried to check which slurm was using:
```
rpm -qa | grep slurm
slurm-example-configs-20.11.8-1.el7.x86_64
```

The shell script (I activated conda environment before submitting):
```
#!/bin/bash
#SBATCH -J 2LiGa111 # Job name
#SBATCH -n 60 # Number of total cores
#SBATCH -N 1 # Number of nodes
#SBATCH --time=02-00:00
#SBATCH -A venkvis_gpu
#SBATCH -p gpu
##SBATCH --nodelist=c024,c025
#SBATCH --mem-per-cpu=0 # Memory pool for all cores in MB
#SBATCH -e outputs_slurm/%x_%A_%a.err #change the name of the err file
#SBATCH -o outputs_slurm/%x_%A_%a.out # File to which STDOUT will be written %j is the job #
#SBATCH --mail-type=ALL # Type of email notification- BEGIN,END,FAIL,ALL
#SBATCH --mail-user=kianpu20@cmu.edu # Email to which notifications will be sent

echo "Job started on `hostname` at `date`

mpiexec -n 60 gpaw python ads_calc.py 2 4

echo " "
echo "Job Ended at `date`"
```

**Comment by abillscmu** [comment]
This is not the same issue, please open a new one

**Comment by kianpu34593** [comment]
Oh sorry, will open a new one. But I checked "what happened" section and found the same error message.
![Screen Shot 2022-05-26 at 9 22 29 AM](https://user-images.githubusercontent.com/59935793/170395713-ad8078a0-d214-49ae-a7ce-8fe6a107298d.png)

**Comment by abillscmu** [comment]
@kianpu34593 is the job running? I get that error too but I ignore it because my job runs anyway

**Comment by jboyd23** [comment]
@abillscmu in my case no error or log files are even being generated as nothing even makes it to the queue. I tried updating the mpirun line of my bash with srun but it has not made any difference. I still just get the bash error and nothing makes it to the queue.

**Comment by abillscmu** [comment]
@jboyd23 the "spank" error is not stopping people from running afaik. If you can give an MWE, I can help you more. Otherwise I have to rely on the errors in the logs in the scripts you put here. And in the script you put here, the error log shows that `FFTW` is unable to be found. It also clearly shows that the job is entering the queue and running. 


Error:
```
/home/jboyd/lammps-29Sep2021/build/lmp: error while loading shared libraries: libfftw3.so.3: cannot open shared object file: No such file or directory
/home/jboyd/lammps-29Sep2021/build/lmp: error while loading shared libraries: libfftw3.so.3: cannot open shared object file: No such file or directory
/home/jboyd/lammps-29Sep2021/build/lmp: error while loading shared libraries: libfftw3.so.3: cannot open shared object file: No such file or directory
/home/jboyd/lammps-29Sep2021/build/lmp: error while loading shared libraries: libfftw3.so.3: cannot open shared object file: No such file or directory
/home/jboyd/lammps-29Sep2021/build/lmp: error while loading shared libraries: libfftw3.so.3: cannot open shared object file: No such file or directory
/home/jboyd/lammps-29Sep2021/build/lmp: error while loading shared libraries: libfftw3.so.3: cannot open shared object file: No such file or directory
/home/jboyd/lammps-29Sep2021/build/lmp: error while loading shared libraries: libfftw3.so.3: cannot open shared object file: No such file or directory
/home/jboyd/lammps-29Sep2021/build/lmp: error while loading shared libraries: libfftw3.so.3: cannot open shared object file: No such file or directory
--------------------------------------------------------------------------
Primary job  terminated normally, but 1 process returned
a non-zero exit code. Per user-direction, the job has been aborted.
--------------------------------------------------------------------------
/home/jboyd/lammps-29Sep2021/build/lmp: error while loading shared libraries: libfftw3.so.3: cannot open shared object file: No such file or directory
/home/jboyd/lammps-29Sep2021/build/lmp: error while loading shared libraries: libfftw3.so.3: cannot open shared object file: No such file or directory
/home/jboyd/lammps-29Sep2021/build/lmp: error while loading shared libraries: libfftw3.so.3: cannot open shared object file: No such file or directory
/home/jboyd/lammps-29Sep2021/build/lmp: error while loading shared libraries: libfftw3.so.3: cannot open shared object file: No such file or directory
/home/jboyd/lammps-29Sep2021/build/lmp: error while loading shared libraries: libfftw3.so.3: cannot open shared object file: No such file or directory
/home/jboyd/lammps-29Sep2021/build/lmp: error while loading shared libraries: libfftw3.so.3: cannot open shared object file: No such file or directory
/home/jboyd/lammps-29Sep2021/build/lmp: error while loading shared libraries: libfftw3.so.3: cannot open shared object file: No such file or directory
/home/jboyd/lammps-29Sep2021/build/lmp: error while loading shared libraries: libfftw3.so.3: cannot open shared object file: No such file or directory
/home/jboyd/lammps-29Sep2021/build/lmp: error while loading shared libraries: libfftw3.so.3: cannot open shared object file: No such file or directory
/home/jboyd/lammps-29Sep2021/build/lmp: error while loading shared libraries: libfftw3.so.3: cannot open shared object file: No such file or directory
/home/jboyd/lammps-29Sep2021/build/lmp: error while loading shared libraries: libfftw3.so.3: cannot open shared object file: No such file or directory
/home/jboyd/lammps-29Sep2021/build/lmp: error while loading shared libraries: libfftw3.so.3: cannot open shared object file: No such file or directory
/home/jboyd/lammps-29Sep2021/build/lmp: error while loading shared libraries: libfftw3.so.3: cannot open shared object file: No such file or directory
/home/jboyd/lammps-29Sep2021/build/lmp: error while loading shared libraries: libfftw3.so.3: cannot open shared object file: No such file or directory
/home/jboyd/lammps-29Sep2021/build/lmp: error while loading shared libraries: libfftw3.so.3: cannot open shared object file: No such file or directory
/home/jboyd/lammps-29Sep2021/build/lmp: error while loading shared libraries: libfftw3.so.3: cannot open shared object file: No such file or directory
/home/jboyd/lammps-29Sep2021/build/lmp: error while loading shared libraries: libfftw3.so.3: cannot open shared object file: No such file or directory
/home/jboyd/lammps-29Sep2021/build/lmp: error while loading shared libraries: libfftw3.so.3: cannot open shared object file: No such file or directory
/home/jboyd/lammps-29Sep2021/build/lmp: error while loading shared libraries: libfftw3.so.3: cannot open shared object file: No such file or directory
/home/jboyd/lammps-29Sep2021/build/lmp: error while loading shared libraries: libfftw3.so.3: cannot open shared object file: No such file or directory
--------------------------------------------------------------------------
mpirun detected that one or more processes exited with non-zero status, thus causing
the job to be terminated. The first process to do so was:

  Process name: [[39785,1],1]
  Exit code:    127
--------------------------------------------------------------------------
```

**Comment by jboyd23** [comment]
If I understand this correctly I believe the gcc/11.2.0 is missing libraries.

**Comment by abillscmu** [comment]
Ok, so this is a very different issue. Do please use your own account though. If you need something from `pyadav`, you can copy that user's bashrc or path or get them to help you.

Please send the script, I just checked, and the spack glibcxx def has 3.4.20, so there must be an error in the script. You can compile on the headnode btw.

**Comment by abillscmu** [comment]
Also if you do 3 backticks:
```
ur messages as code look much better
```

**Comment by jboyd23** [comment]
@abillscmu 
```#!/bin/bash
#SBATCH -J n_1_s1 # Job name
#SBATCH -n 28 # Number of total cores
#SBATCH -N 1 # Number of nodes
#SBATCH -A barati
#SBATCH -p cpu
#SBATCH --mem-per-cpu=2000 # Memory pool for all cores in MB
#SBATCH -e /home/pyadav/Boyd_Sim/Barepore_Current/20/n_1/S1/error_lammps.err
#SBATCH -o /home/pyadav/Boyd_Sim/Barepore_Current/20/n_1/S1/out_lammps.out
#SBATCH --mail-type=END,FAIL,BEGIN # Type of email notification- BEGIN,END,FAIL,ALL
#SBATCH --mail-user=pyadav@andrew.cmu.edu # Email to which notifications will be sent
#SBATCH -t 7-0:00
echo "Job started on `hostname` at `date`"
module load openmpi/4.1.3
module load gcc/11.2.0
srun --mpi=pmix -n 28 /home/pyadav/lammps/src/lmp_mpi < Phos_memb20A2_Complete.in


echo " "
echo "Job Ended at `date`"

**Comment by abillscmu** [comment]
Great! Now we have something to work with! I can reproduce this.

What happens if you just don't load `gcc`

**Comment by abillscmu** [comment]
That probably won't work. Try replacing `module load gcc` with `spack load gcc`

**Comment by jboyd23** [comment]
```Error: No installed spec matches the hash: '11.2.0
![Screenshot (35)](https://user-images.githubusercontent.com/94718071/170529366-2cc04221-48c1-46a3-ba33-88f2202cdc80.png)
'

**Comment by jboyd23** [comment]
Its telling me that no hash matches gcc/11.2.0 but if I use ```spack find``` its there
``` Error: No installed spec matches the hash: '11.2.0'

**Comment by abillscmu** [comment]
`gcc@11.2.0`

**Comment by abillscmu** [comment]
But I think you may need to recompile lammps

**Comment by jboyd23** [comment]
@abillscmu 

After further analysis I determined that ```gcc/11.2.0``` wasn't properly loaded. However, when I tried to manually load gcc ```module load gcc/11.2.0``` it is unrecognized. Is this an issue on your end? As I would not be able to recompile lammps until gcc was recognized? 

```[pyadav@coe ~]$ module load gcc/11.2.0
Lmod has detected the following error:  The following module(s) are unknown: "mpfr/4.1.0" "gmp/6.2.1"
"mpc/1.1.0" "zlib/1.2.11" "zstd/1.5.0"

Please check the spelling or version number. Also try "module spider ..."
It is also possible your cache file is out-of-date; it may help to try:
  $ module --ignore_cache load "mpfr/4.1.0" "gmp/6.2.1" "mpc/1.1.0" "zlib/1.2.11" "zstd/1.5.0"

**Comment by abillscmu** [comment]
yeah, I'm getting the same thing. You should be able to do `spack load gcc@11.2.0` and it should work.

**Comment by jboyd23** [comment]
@abillscmu 
 So gcc magically loads now using ```module load``` and ```spack load``` so everything is working properly now thank you so much for all of your help

**Comment by awadell1** [comment]
Apologies for the issue, the gcc module was out of date. It has been updated
